pyuvdata/uvdata.py

```diff
@@ -1583,6 +1583,57 @@ def _key2inds(self, key):
                 raise KeyError('Polarization {pol} not found in data.'.format(pol=key[2]))
         return (blt_ind1, blt_ind2, pol_ind)
 
+
+    def _smart_slicing(self, data, ind1, ind2, indp):
+        """
+        Method for quickly picking out the relevant section of data for get_data or get_flags
+
+        Args:
+            data: 4-dimensional array in the format of self.data_array
+            ind1: list with blt indices for antenna pair (e.g. from self._key2inds)
+            ind2: list with blt indices for conjugate antenna pair. (e.g. from self._key2inds)
+            indp: list with polarization indices (e.g. from self._key2inds)
+
+        Returns:
+            out: copy (or if possible, view) of relevant section of data as an numpy array
```

Comment:
But Josh I feel like you just described exactly why we need a warning - the user wants the data as fast as possible, but sometimes it will be a copy and sometimes not.
One alternative I'm looking into is read-only views. We could return arrays that cannot be changed, unless the user manually changes that setting. thoughts?